### ISSUE TITLE
Bad gradient for pool with padding

### ISSUE THREAD

**Issue by denizyuret** [FluxML/NNlib.jl#133]
Unless I am doing something wrong:
```
using NNlib

x = reshape([(1.:9.)...], 3, 3, 1, 1)
pdims = PoolDims(size(x), (2,2), padding = (1,1), stride = (2,2))
y = maxpool(x, pdims)
dy = y .* 0 .+ 1
dx = ∇maxpool(dy, y, x, pdims)

display(x)
display(y)
display(dy)
display(dx)
```
gives
```
3×3×1×1 Array{Float64,4}:                                                                                                     
[:, :, 1, 1] =                                                                                                                
 1.0  4.0  7.0                                                                                                                
 2.0  5.0  8.0                                                                                                                
 3.0  6.0  9.0                                                                                                                
2×2×1×1 Array{Float64,4}:                                                                                                     
[:, :, 1, 1] =                                                                                                                
 1.0  7.0                                                                                                                     
 3.0  9.0                                                                                                                     
2×2×1×1 Array{Float64,4}:                                                                                                     
[:, :, 1, 1] =                                                                                                                
 1.0  1.0                                                                                                                     
 1.0  1.0                                                                                                                     
3×3×1×1 Array{Float64,4}:                                                                                                     
[:, :, 1, 1] =                                                                                                                
 1.0  0.0  0.0                                                                                                                
 0.0  0.0  0.0                                                                                                                
 0.0  0.0  1.0                                                                                                                
```
The gradients for 3 and 7 should not be 0.

**Comment by MikeInnes** [comment]
@staticfloat or @avik-pal would have looked at this code most recently, so might be able to help debug.

**Comment by staticfloat** [comment]
Yep, that's a bug.  Luckily, it was very easy to fix.  It was masked within our (non-exhaustive) tests because our tests did not include a case with padding where the window size was an uneven multiple of the input size.  I've included a fix and a new test in #134
